src/battle_ai_switch_items.c

```diff
@@ -897,37 +1079,774 @@ static u32 GetBestMonDmg(struct Pokemon *party, int firstId, int lastId, u8 inva
+            passiveDamage = maxHP / 8;
+            if(passiveDamage == 0)
+                passiveDamage = 1;
+        }
+    }
+    return passiveDamage;
+}
+
+// Gets one turn of status damage
+static u32 GetSwitchinStatusDamage(u32 battler)
+{
+    u8 defType1 = AI_DATA->switchinCandidate.battlemon.type1, defType2 = AI_DATA->switchinCandidate.battlemon.type2;
+    u8 tSpikesLayers = gSideTimers[GetBattlerSide(battler)].toxicSpikesAmount;
+    u16 heldItemEffect = gItems[AI_DATA->switchinCandidate.battlemon.item].holdEffect;
+    u32 status = AI_DATA->switchinCandidate.battlemon.status1, ability = AI_DATA->switchinCandidate.battlemon.ability, maxHP = AI_DATA->switchinCandidate.battlemon.maxHP;
+    u32 statusDamage = 0;
+
+    // Status condition damage
+    if ((status != 0) && AI_DATA->switchinCandidate.battlemon.ability != ABILITY_MAGIC_GUARD) 
+    {
+        if (status & STATUS1_BURN) 
+        {
+            #if B_BURN_DAMAGE >= GEN_7
+                statusDamage = maxHP / 16;
+            #else
+                statusDamage = maxHP / 8;
+            #endif
+            if(ability == ABILITY_HEATPROOF)
+                statusDamage = statusDamage / 2;
+            if (statusDamage == 0)
+                statusDamage = 1;
+        }
+        else if (status & STATUS1_FROSTBITE)
+        {
+            #if B_BURN_DAMAGE >= GEN_7
+                statusDamage = maxHP / 16;
+            #else
+                statusDamage = maxHP / 8;
+            #endif
+            if (statusDamage == 0)
+                statusDamage = 1;
+        }
+        else if ((status & STATUS1_POISON) && ability != ABILITY_POISON_HEAL)
+        {
+            statusDamage = maxHP / 8;
+            if (statusDamage == 0)
+                statusDamage = 1;
+        }
+        else if ((status & STATUS1_TOXIC_POISON) && ability != ABILITY_POISON_HEAL)
+        {
+            if ((status & STATUS1_TOXIC_COUNTER) != STATUS1_TOXIC_TURN(15)) // not 16 turns
+                AI_DATA->switchinCandidate.battlemon.status1 += STATUS1_TOXIC_TURN(1);
+            statusDamage *= AI_DATA->switchinCandidate.battlemon.status1 & STATUS1_TOXIC_COUNTER >> 8;
+            if (statusDamage == 0)
+                statusDamage = 1;
+        }
+    }
+
+    // Apply hypothetical poisoning from Toxic Spikes, which means the first turn of damage already added in GetSwitchinHazardsDamage
+    // Do this last to skip one iteration of Poison / Toxic damage, and start counting Toxic damage one turn later.
+    if (tSpikesLayers != 0 && ((defType1 != TYPE_POISON && defType2 != TYPE_POISON
+        && ability != ABILITY_IMMUNITY && ability != ABILITY_POISON_HEAL
+        && status == 0
+        && !(heldItemEffect == HOLD_EFFECT_HEAVY_DUTY_BOOTS 
+            && (((gFieldStatuses & STATUS_FIELD_MAGIC_ROOM) || ability == ABILITY_KLUTZ)))
+        && heldItemEffect != HOLD_EFFECT_CURE_PSN && heldItemEffect != HOLD_EFFECT_CURE_STATUS
+        && defType1 != TYPE_FLYING && defType2 != TYPE_FLYING
```

Comment:
Since this checks if pokemon is grounded the same way as your spikes check, it could be a separate function not to duplicate code.